types/novnc-core/index.d.ts

```diff
@@ -0,0 +1,112 @@
+// Definitions by: Ken Smith <https://github.com/smithkl42/>
+//                 BendingBender <https://github.com/BendingBender>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import Display from './lib/display';
+import { Keyboard, Mouse } from './lib/input/devices';
+
+export default class RFB {
+	constructor(defaults?: NvRFBDefaults);
+    get_target(): HTMLCanvasElement;
+    set_target(target: HTMLCanvasElement): void;
+    get_focusContainer(): HTMLElement;
+    set_focusContainer(container: HTMLElement): void;
+    get_encrypt(): boolean;
+    set_encrypt(encrypt: boolean): void;
+    get_local_cursor(): boolean;
+    set_local_cursor(localCursor: boolean): void;
+    get_shared(): boolean;
+    set_shared(shared: boolean): void;
+    get_view_only(): boolean;
+    set_view_only(viewOnly: boolean): void;
+    get_xvp_password_sep(): string;
+    set_xvp_password_sep(xvpPasswordSep: string): void;
+    get_disconnectTimeout(): number;
+    set_disconnectTimeout(disconnectTimeout: number): void;
+    get_wsProtocols(): string[];
+    set_wsProtocols(wsProtocols: string[]): void;
+    get_repeaterID(): string;
+    set_repeaterID(repeaterID: string): void;
+    get_viewportDrag(): boolean;
+    set_viewportDrag(viewportDrag: boolean): void;
+    get_onUpdateState(): (rfb: this, state: NvConnectionState, oldstate: NvConnectionState) => void;
+    set_onUpdateState(handler: (rfb: this, state: NvConnectionState, oldstate: NvConnectionState) => void): void;
+    get_onNotification(): (rfb: this, msg: string, level: 'normal' | 'warn' | 'error', options?: { [key: string]: any }) => void;
+    set_onNotification(handler: (rfb: this, msg: string, level: 'normal' | 'warn' | 'error', options?: { [key: string]: any }) => void): void;
+    get_onDisconnected(): (rfb: this, reason?: string) => void;
+    set_onDisconnected(handler: (rfb: this, reason?: string) => void): void;
+    get_onPasswordRequired(): (rfb: this, msg?: string) => void;
+    set_onPasswordRequired(handler: (rfb: this, msg?: string) => void): void;
+    get_onClipboard(): (rfb: this, text: string) => void;
+    set_onClipboard(handler: (rfb: this, text: string) => void): void;
+    get_onBell(): (rfb: this) => void;
+    set_onBell(handler: (rfb: this) => void): void;
+    get_onFBUReceive(): (rfb: this, fbu: NvFBU) => void;
+    set_onFBUReceive(handler: (rfb: this, fbu: NvFBU) => void): void;
+    get_onFBUComplete(): (rfb: this, fbu: NvFBU) => void;
+    set_onFBUComplete(handler: (rfb: this, fbu: NvFBU) => void): void;
+    get_onFBResize(): (rfb: this, width: number, height: number) => void;
+    set_onFBResize(handler: (rfb: this, width: number, height: number) => void): void;
+    get_onDesktopName(): (rfb: this, name: string) => void;
+    set_onDesktopName(handler: (rfb: this, name: string) => void): void;
+    get_onXvpInit(): (version: number) => void;
+    set_onXvpInit(handler: (rfb: this, name: string) => void): void;
+	get_display(): Display;
```

Comment:
Why do you have these get_ and set_ methods? https://github.com/larryprice/novnc-core#usage says the only public methods of this class are: 

> /*
>  An rfb has the following public methods (see lib/rfb.js for more details):
>   * connect(host, port, password, path)
>   * disconnect()
>   * sendPassword(passwd)
>   * sendCtrlAltDel()
>   * xvpOp(ver, op)
>   * xvpShutdown()
>   * xvpReboot()
>   * xvpReset()
>   * sendKey(keysym, code, down)
>   * clipboardPasteFrom(text)
>   * requestDesktopSize(width, height)
> */